File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02900/p02900_num9_parsed.py
# Program to find the maximum number of divisors that can be chosen
# to satisfy the condition that any two chosen divisors must be coprime.

# Function to find the number of divisors of a number
def divisors_count(num):
    count = 0
    for i in range(1, int(num**0.5) + 1):
        if num % i == 0:
            count += 1
            if i != num // i:
                count += 1
    return count


# Function to find gcd (greatest common divisor) of two numbers
def gcd(a, b):
    while b:
        a, b = b, a % b
    return a

# Input
A, B = map(int, input().split())

# Find the gcd of A and B
gcd_value = gcd(A, B)

# If gcd is 1, it means A and B are coprime,
# so we can choose only 1 common divisor which is 1 itself.
if gcd_value == 1:
    print(1)
else:
    # Count the number of divisors of the gcd
    divisors = divisors_count(gcd_value)
    print(divisors)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02900/p02900_num9_parsed.cpp
#include <iostream>
#include <vector>

using namespace std;

// Function to calculate the greatest common divisor (GCD) of two numbers
long long gcd(long long a, long long b) {
    if (b == 0) {
        return a;
    }
    return gcd(b, a % b);
}

// Function to calculate the number of positive common divisors of A and B
int countCommonDivisors(long long A, long long B) {
    long long d = gcd(A, B);
    vector<long long> divisors;
    
    // Finding all the divisors of d
    for (long long i = 1; i * i <= d; i++) {
        if (d % i == 0) {
            divisors.push_back(i);
            if (i != d / i) {
                divisors.push_back(d / i);
            }
        }
    }
    
    int n = divisors.size();
    int maxDivisors = 0;
    
    // Loop to check if any two divisors are coprime
    for (int i = 0; i < n; i++) {
        bool coprime = true;
        for (int j = i + 1; j < n; j++) {
            if (gcd(divisors[i], divisors[j]) != 1) {
                coprime = false;
                break;
            }
        }
        if (coprime) {
            // Incrementing the count if the two divisors are coprime
            maxDivisors++;
        }
    }
    
    return maxDivisors;

}

int main() {
    // Read the input values
    long long A, B;
    cin >> A >> B;
    
    // Call the function to calculate the maximum number of divisors
    int result = countCommonDivisors(A, B);
    
    // Print the result
    cout << result << endl;

    return 0;
}